add mdx authless doc (#10)
miamtech/mealz-documentation: docs/shared/advanced/AuthLess.mdx

export var platform 

import AndroidAuthlessConfig from '@site/docs/android/advanced/_shared/_user/_AuthlessConfig.mdx'
import AndroidLoginRedirectionConfig from '@site/docs/android/advanced/_shared/_user/_LoginRedirectionConfig.mdx'

import IosAuthlessConfig from '@site/docs/ios/advanced/_shared/_user/_AuthlessConfig.mdx'
import IosLoginRedirectionConfig from '@site/docs/ios/advanced/_shared/_user/_LoginRedirectionConfig.mdx'

export function AuthlessConfig(props){
    if(props.platform == "android") { return <AndroidAuthlessConfig /> } else return  <IosAuthlessConfig />
}

export function LoginRedirectionConfig(props){
    if(props.platform == "android") { return <AndroidAuthlessConfig /> } else return  <IosLoginRedirectionConfig />
}

> AuthLess mode alow mealz to create an anonymous user and to find back previous list of none connected user.
> Once user connect on your app Mealz will merge anonymous user's list to the conected user's one

Reviewer: ```suggestion
AuthLess mode allows Mealz to create an anonymous user and to find previous lists of non-connected users.
Once the user connects to your app Mealz will merge anonymous user into the connected user.
```

:::info
AuthLess user feature is only available since version 4.1 and above 
:::

You can enable AuthLess feature in mealz initialisation

<AuthlessConfig platform= {props.platform}/>

:::info
**isAnonymousModeEnabled** is defaulted to **false**
:::

> You can also provide to Mealz a redirection function to your login page. Then we'll use this function when user connection become mandatory. In this case there is no anonymous user but you can still show mealz recipe even if mealz don't have all information it needs

Reviewer: ```suggestion
You can also provide Mealz a redirection function to your login page. We will use this function when connecting the user connection becomes mandatory, like adding the products to the basket. Even without supporting anonymous users, we can display Mealz Recipes on your apps, with CTAs requiring a user sign in. 
```
